### ISSUE TITLE
The voice input device stopped working after updating and resetting the settings.

### ISSUE THREAD

**Issue by PstalDude** [ValveSoftware/steam-for-linux#5617]
The voice input device stopped working after updating and resetting the settings.

Steam client version: 1533766730
Distribution: Linux mint 19
Opted into Steam client beta: No
Checked for system updates: Yes

About a week ago, a new Steam update came in, where the voice input settings menu was cut from the Steam settings menu. At this point, Steam settings were reset by default (this often happens for reasons unknown to me), along with them, the voice input device was reset by default. I tried to set up the voice input device through the settings of the new chat (where there are two devices: "Default", which does nothing, and "Built-in Analog Audio", which outputs the sound from the speakers to voice chat). And also through "Steam Voice Settings" (went through Dota 2), where only "PulseAudio default" is installed. None of the options output sound from the microphone. I entered the pavucontrol where the voice input device for Steam was set to "Monitor of Built-in Analog Audio" (outputs the sound from the speakers). I tried to change this option to "Built-in Analog Audio"(outputs the sound from the microphone), but it did not work out for reasons unknown to me. For other programs with voice input (for example, Discord), the voice input device changed freely from "Monitor of Built-in Analog Audio" to "Built-in Analog Audio". I tried to reinstall Steam, and also rolled back the pulseaudio package (although it was not updated lately) but to no avail.

**Comment by u83rmensch** [comment]
I also have this issue. really wish valve would just update their individual titles to not rely on the microphone input from steam and just have their own settings with in the game. 

that said I've settled on a workaround for this. I use kde so it might be different in other  DE's however I just disable all other audio inputs/outputs (primarily my onboard audio chipset) except my headset before launching dota. This forces the game to only have one audio source to pick from. Its a rough solution but works just fine.

**Comment by cuevasclemente** [comment]
I can do a process similar to this to get voice working, but the problem with this workaround is that if I later wish to use Steam In-Home Streaming, it creates problems because the in-home streaming relies on using a monitor of some stereo mix that I have. If I disable it, it appears to permanently stop transmitting audio until I reset my computer.

**Comment by shpinog** [comment]
I`am have problem too

**Comment by Nemoder** [comment]
This is a real annoyance for me as well, if a game is running you can use shift-tab to access recording options but if the game is already loaded with the wrong device it doesn't seem to allow being changed.

**Comment by igorrates** [comment]
Is there any update on this issue? Is really annoying not being able to change my device since I use custom configuration to be able to stream

**Comment by AdamSzopa** [comment]
I believe I have the same or very similar problem.

I had one pair of USB Phillips headphones, and everything worked fine. Now I have a new pair, and Steam seems to be stuck internally on the old ones. Let me explain.

Within Dota 2, I can select the proper audio device from the list - either "system default" or selecting the headphones work (the headphone are selected as default in my OS) - and I can hear everything fine, but the mic is not recording. In my system tray I see a notification that "Steam" is using the microphone.

In Dota 2 settings, there is a... link (?) to the general Steam audio settings, which launches in the overlay. As far as I know I cannot access this panel from anywhere else, it's not in the general Steam settings anymore. I wish I could access it from elsewhere, because the overlay is buggy, and doesn't register 90% of my clicks, and the windows seems offset from what I see on the screen.

In this window I can see that the input device is "USB Philips from alsa" or something, so the wrong ones - they are not connected. Clicking "search for devices" does nothing. "Change device" is grayed out (I assume it's only enabled when more then 1 device is found). Test mic button works, but nothing is detected.

Then there is the 3rd window for audio settings, in the new friend list - there also I can change the devices but THAT test mic dialog also shows that nothing is working.

Note that for example Discord works with the new device fine.

**Comment by SgtMarv** [comment]
so has this ever been fixed? because I'm still having the issue?

**Comment by shpinog** [comment]
I have no more problems after switching to jack.
I use a bunch of alsa-pulse-jack through cadence.

The whole problem is which device you have first in pavucontrol, while sometimes steam doesn't matter which one is the default. Only one jack sink device remains through the jack in pulse, so the problem is solved.

**Comment by Kamiskub** [comment]
Steam has bug which leads to not being able to choose input device from steam settings so it chooses windows default one. To make your steam voice chat work again you need to change default COMMUNICATION device to your proper mic input. In my case after installing Virtuall Cable all apps with custom settings kept working properly as I could choose which input to use but Steam didn't and I was stuck with Virtual Cable as Input device. 
Tl;dr make sure your default input for other apps is also your windows default communication device.
![solution](https://user-images.githubusercontent.com/60986736/74373814-78eb0200-4ddd-11ea-9721-6982ca314303.PNG)

**Comment by Kamiskub** [comment]
Steam is bugged and because you can't choose default input device from steam you need to do it from control panel. Make sure your default for windows input device is the only one with green marker.
![solution](https://user-images.githubusercontent.com/60986736/74374659-201c6900-4ddf-11ea-8537-514cee5ba1c3.PNG)

**Comment by shpinog** [comment]
@Kamiskub

Distribution: Linux mint 19. Repo named - steam-for-linux. Good solution for windows.

**Comment by u83rmensch** [comment]
> Steam is bugged and because you can't choose default input device from steam you need to do it from control panel. Make sure your default for windows input device is the only one with green marker.
> ![solution](https://user-images.githubusercontent.com/60986736/74374659-201c6900-4ddf-11ea-8537-514cee5ba1c3.PNG)

bro, you realize you're on a linux support page yeah? Windows help doesnt do dick here.

**Comment by Mameng57** [comment]
idk whether this is pulseaudio's fault or other input software that the distro uses.
but steam also doesn't even showing the option to debug this mess. it works on other recording apps though

**Comment by mbkv** [comment]
I think I've figured out the issue. adding in

```
drivers = pulse,alsa
```

to my `.alsoftrc` fixed it. this started happening when I started using hrtf but I never made the connection until I looked up more settings for alsoft

**Comment by AlexandreBonneau** [comment]
For information, the microphone is not recognized (apart 'Default') in the current steam and steam beta version.
This is under Debian testing (with pipewire `0.3.59-1`).
